Improving battery signals

erikbosch opened the pull request:
This PR tries to remove some ambiguities I see. Fixes #354 

I am not saying that the proposed changes are "right". I am happy to receive counter-proposals if someone has a better proposal, or if we in some cases shall keep the ambiguity as an OEM variation point.

erikbosch commented on spec/Powertrain/Battery.vspec:
```diff
@@ -127,7 +127,7 @@ Charging.ChargePortFlap:
   datatype: boolean
   type: sensor
-  description: Signal indicating if charge plug is connected or not.
+  description: Signal indicating if a charging cable is connected to the charge plug or not.
```
I am not 100% sure on the intention of this signal. If the intention is to focus on the cable or on the connection to a charger? I.e. if you connect a cable to a vehicle, but do not connect the other end to a charger, shall this signal then anyway shall true? Or shall the focus rather be on if the vehicle is connected to a functional/powered charger?

erikbosch replied:
Meeting: Change name to "IsChargingCableConnected"

erikbosch commented on spec/Powertrain/Battery.vspec:
```diff
@@ -185,12 +187,12 @@ Charging.Timer.Time:
   datatype: string
   type: actuator
-  enum: [ 'starttime', 'endtime', 'depaturetime' ]
+  enum: [ 'inactive', 'starttime', 'endtime', 'departuretime' ]
```
Here I assume that a specific element to indicate that timer-based charging is deactivated could make sense if you want to cancel planned charging, rather than setting Charging.Timer.Time to 0.

I corrected spelling for departuretime as well, but do not know if it makes sense to keep it as long as we do not have any "departuretime" signal in VSS

danielwilms commented on spec/Powertrain/Battery.vspec:
```diff
@@ -145,7 +145,7 @@ Charging.Mode:
 Charging.Status:
```
* @danielwilms will check internally (@adobekan, or you?) 
* other proposal: `isCharging` instead of `Status`

erikbosch commented on spec/Powertrain/Battery.vspec:
```diff
@@ -185,12 +187,12 @@ Charging.Timer.Time:
   datatype: uint32
```
Meeting: use string, format according to ISO standard 8601

erikbosch commented:
Reworked based on comments from last meeting, changes performed:

- Renamed boolean signals to start with "Is"
- Align descriptions so they alway end with a period (`.`)
- Changed timer to use string value. Also proposing to remove the `departuretime` alternative as we do not have any signal for that in VSS

adobekan commented:
@danielwilms If needed, I can check next week. Somehow i missed the notification.

erikbosch commented:
Updated `Charging.Timer.Time` to clarify that time shall be given in UTC.

An ISO 8601 timestamp can exclude time zone information, i.e. referring to "local time zone" which would be ambiguous.  I also do not really see any value in specifying time with offset from UTC. This is aligned with VISS that also require UTC (Z) time.

danielwilms commented:
meeting 2021-12-14 approved
